backend/typescript/services/implementations/donorService.ts

```diff
@@ -173,6 +166,13 @@ class DonorService implements IDonorService {
         );
         throw error;
       }
+      const numDestroyed: number = await Donor.destroy({
```

Comment:
we need delete schedules before we delete donors cuz if we dont, we will get foreign key constraint error. Same for volunteers, which is why I added this change in both of the delete functions.